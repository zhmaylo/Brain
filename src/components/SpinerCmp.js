import React from 'react';
import { View, StyleSheet, Text, Image, } from 'react-native';
import { SPINER_GIF, SPINER_MES } from './../constants/spinerConst';

// spinner component
// volume - current spiner volume
// max - max spiner volume
export const SpinerСmp = () => {
    // console.log("SpinerСmp. volume => ", volume);
    // console.log("SpinerСmp. max => ", max);
    return (
        <View style={styles.container}>
            <Image
                style={styles.image}
                source={SPINER_GIF}
            />
            <Text>{SPINER_MES}</Text>
        </View>)
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: "center",
        alignItems: "center",
    },
    image: {
        width: 30,
        height: 30,
    },

})
