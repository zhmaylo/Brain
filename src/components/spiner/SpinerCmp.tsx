import React from 'react';
import { View, StyleSheet, Text, ActivityIndicator } from 'react-native';
import { SPINER_MES } from '../../constants/spinerConst';

// spinner component
// volume - current spiner volume
// max - max spiner volume
export const SpinerСmp = () => {
    // console.log("SpinerСmp. volume => ", volume);
    // console.log("SpinerСmp. max => ", max);
    return (
        <View style={styles.container}>
            <ActivityIndicator size="large" color="#000000" />
            <Text>{SPINER_MES}</Text>
        </View>)
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: "center",
        alignItems: "center",
    },

})
