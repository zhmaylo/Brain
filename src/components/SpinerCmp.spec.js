
import React from 'react';
import renderer from 'react-test-renderer';
import { SpinerСmp } from './SpinerCmp'

test('renders correctly', () => {
    
    const tree = renderer.create(
        SpinerСmp (25, 200))
        .toJSON();
    expect(tree).toMatchSnapshot();
});



// tasca - testAsyncCallback Creates Async Callback test
// taspr - testAsyncPromise Creates Aysnc Promise test
// tasaw - testAsyncAwait Creates Async Await test
// tsnsh - testSnapShot Creates Snapshot test
// tshre - testShallowRenderer Creates Shallow Renderer test