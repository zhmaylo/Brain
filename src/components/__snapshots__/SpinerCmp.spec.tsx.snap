// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#000000"
    hidesWhenStopped={true}
    size="large"
  />
  <Text>
    Загрузка...
  </Text>
</View>
`;
